File: DoiTuyen/Practice/Ksat tháng 4/Test 20.04.2024/MATONG/matong.py
fi = open("matong.inp","r")
n = int(fi.readline())
a = list(map(int,fi.readline().split()))
fi.close()

def f(x):
    if (x == 1): return 1
    i = 2
    count_div = 1
    while (i*i <= x):
        d = 0
        while (x % i == 0):
            d += 1
            x = x // i
        count_div *= d + 1
        i += 1
    if (x > 1): count_div *= 2
    return count_div

ans = 0
for x in a:
    ans += x * f(x)
fo = open("matong.out","w")
fo.write(str(ans))
fo.close()

File: DoiTuyen/Practice/Ksat tháng 4/Test 20.04.2024/MATONG/matong.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
int n;
ll ans;
ll f(ll x)
{
    if (x == 1) return 1;
    ll count_div = 1;
    ll i = 2;
    while (i*i <= x)
    {
        ll d = 0;
        while (x % i == 0)
        {
            d++;
            x = x / i;
        }
        count_div *= d + 1;
        i++;
    }
    if (x > 1) count_div *= 2;
    return count_div;
}
int main()
{
    freopen("matong.inp","r",stdin);
    freopen("matong.out","w",stdout);
    cin >> n;
    for (int i=1; i<=n; i++)
    {
        ll x;
        cin >> x;
        ans += x * f(x);
    }
    cout << ans;
    return 0;
}
